### PR TITLE
Make views navigable via search shortcuts to simplify automated testing.

### PR DESCRIPTION
**Pull request by chuckharmston**
r? @ngokevin @mstriemer @spasovski 

This was a request from @bobsilverberg that basically does what we do for `:debug` with the majority of our argument-less views.

### CODE REVIEW

**Comment by chuckharmston** [comment]
I realized that this overrides the `:new` and `:popular` search shortcuts. I'm going to refactor a bit to avoid that.

File: src/media/js/views/search.js [hunk 1]
```diff
@@ -103,12 +103,28 @@ define('views/search',
...
         e.stopPropagation();
+        // A mapping of query => view name that can be used to navigate to a
+        // specific view via search box. Mostly used for automated testing.
+        var potato_views = {
+            ':categories': 'categories',
+            ':debug': 'debug',
+            ':debug_features': 'debug_features',
+            ':feedback': 'feedback',
+            ':homepage': 'homepage',
+            ':new_apps': 'new',
```
Comment (ngokevin) [inline src/media/js/views/search.js]: Just ```new``` and just ```popular``` would fit with what we call them everywhere else.

Comment (chuckharmston) [inline src/media/js/views/search.js]: :new and :popular are existing sort filters, though.

Comment (ngokevin) [inline src/media/js/views/search.js]: ah didn't see that

**Comment by ngokevin** [comment]
lgtm

**Comment by chuckharmston** [comment]
Thanks!

**Pull request merged**